zrml/court/src/lib.rs

```diff
@@ -16,611 +16,2255 @@
+        CourtIdToMarketIdNotFound,
+        /// The vote item is not valid for this (outcome) court.
+        InvalidVoteItemForOutcomeCourt,
+        /// The vote item is not valid for this (binary) court.
+        InvalidVoteItemForBinaryCourt,
+        /// The appealed vote item is not an outcome.
+        AppealedVoteItemIsNotOutcome,
+        /// The winner vote item is not an outcome.
+        WinnerVoteItemIsNoOutcome,
+        /// The outcome does not match the market outcomes.
+        OutcomeMismatch,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+        fn on_initialize(now: T::BlockNumber) -> Weight {
+            let mut total_weight: Weight = Weight::zero();
+            total_weight = total_weight.saturating_add(Self::handle_inflation(now));
+            total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+            if now >= <RequestBlock<T>>::get() {
+                let future_request = now.saturating_add(T::RequestInterval::get());
+                <RequestBlock<T>>::put(future_request);
+                total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+            }
+            total_weight
+        }
+
+        fn integrity_test() {
+            assert!(!T::BlocksPerYear::get().is_zero(), "Blocks per year assumption changed.");
+        }
+    }
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
-        // MARK(non-transactional): `remove_juror_from_all_courts_of_all_markets` is infallible.
-        #[pallet::weight(T::WeightInfo::exit_court())]
-        pub fn exit_court(origin: OriginFor<T>) -> DispatchResult {
+        /// Join to become a juror, who is able to get randomly selected
+        /// for court cases according to the provided stake.
+        /// The probability to get selected is higher the more funds are staked.
+        /// The `amount` of this call represents the total stake of the juror.
+        /// If the pool is full, the lowest staked juror is removed from the juror pool.
+        ///
+        /// # Arguments
+        ///
+        /// - `amount`: The total stake associated with the joining juror.
+        ///
+        /// # Weight
+        ///
+        /// Complexity: `O(log(n))`, where `n` is the number of jurors in the stake-weighted pool.
+        #[pallet::weight(T::WeightInfo::join_court(T::MaxJurors::get()))]
+        #[transactional]
+        pub fn join_court(
+            origin: OriginFor<T>,
+            amount: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
             let who = ensure_signed(origin)?;
-            let juror = Self::juror(&who)?;
-            Self::remove_juror_from_all_courts_of_all_markets(&who);
-            Self::deposit_event(Event::ExitedJuror(who, juror));
-            Ok(())
+
+            let jurors_len = Self::do_join_court(&who, amount, None)?;
+
+            Self::deposit_event(Event::JurorJoined { juror: who, stake: amount });
+
```

Comment:
It seems to me that the current approach makes more sense, because it sounds confusing to emit `JurorJoined` for the `delegate` dispatchable function.